specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,262 @@
+| `inttoptr`               | Used to cast an `i64` integer value to either a `%Qubit*` or a `%Result*`.                                       | May be used as part of a function call only.                                                          |
+| `getelementptr inbounds` | Used to create an `i8*` to pass a constant string for the purpose of labeling an output value.                   | May be used as part of call to an output recording function only.                                     |
+
+## Data Types and Values
+
+Within the base profile, defining local variables is not supported. Arguments to
+calls correspondingly are expected to be constant values.
+
+Qubits and results are passes as a pointer of type `%Qubit*` and `%Result*`
+respectively, where the pointer itself rather than the memory location
+identifies the qubit or result: For the purpose of passing them as arguments in
+function calls, a 64-bit signed integer value is cast to the appropriate pointer
+type. The signed integer that is case must be in the interval `[0, nrQubits)`
+for `%Qubit*` and `[0, nrResults)` for `%Result*`, where `nrQubits` and
+`nrResults` are the required number of qubits and results defined by entry point
+attributes.
+
+Only the `%Qubit` and `%Result` data types are required to be supported by all
+backends.
+
+Integers and double-precision floating point numbers are available as in full
+QIR; however, computations using these numeric types are not available.
+
+### Qubit and Result Usage
+
+runtime has autonomy over how to represent qubits. also for base profile, the
+runtime is the only entity that needs to know how to interpret the opaque
+pointers (deref or not). that info is captured in the target triple/duo.
+
+Qubits may not be used after they have been measured. Qubits and results need to
+be numbered consecutively, starting at 0. No qpu support needed for measuring
+individual qubits, but backend would support output selection?
+
+## Quantum Instruction Set
+
+For a Quantum Instruction Set to be compatible with the Base Profile, it needs
+to satisfy the following requirements:
+
+- Since the Base Profile doesn't permit to define local variables, all
+  instructions are required to return void.
+
+## Output Recording
+
+Log format is a separate spec. What the i8* can be is a separate spec. Default
+spec for front-ends to compile into. TODO: do string labels need to be zero
+terminated?
+
+The following functions are declared and used to record the program output: |
+Function                  | Signature         | Description |
+|---------------------------|-------------------|-------------| |
+__quantum__rt__initialize_record_output    | `void(i8*)`  | Inserts a marker in
+the output log that contains an identifier for the used labeling scheme. The
```

Comment:
I do not believe the output log has been defined at this point.

Reply:
The document now contains a bunch of clarifications regarding output schemas. I updated all mentioned of "output format" and "log"  to hopefully make it clear and consistent terminology. Please take another look and let me know if anything is unclear.